apps/openchallenges/data-lambda/openchallenges_data_lambda/oc_data_sheet.py

```diff
@@ -0,0 +1,193 @@
+    challenges["headline"] = (
+        challenges["headline"]
+        .astype(str)
+        .apply(lambda x: x[:75] + "..." if len(x) > 80 else x)
+    )
+    challenges["description"] = (
+        challenges["description"]
+        .astype(str)
+        .apply(lambda x: x[:995] + "..." if len(x) > 1000 else x)
+    )
+    challenges.loc[challenges._platform == "Other", "platform"] = None
+    challenges.loc[challenges.start_date == "", "start_date"] = None
+    challenges.loc[challenges.end_date == "", "end_date"] = None
+    challenges.loc[challenges.operation_id == "", "operation_id"] = None
+
+    # Challenge incentive(s)
+    incentives = pd.concat(
+        [
+            df[df.monetary_incentive == "TRUE"][["id", "created_at"]].assign(
+                name="monetary"
+            ),
+            df[df.publication_incentive == "TRUE"][["id", "created_at"]].assign(
+                name="publication"
+            ),
+            df[df.speaking_incentive == "TRUE"][["id", "created_at"]].assign(
+                name="speaking_engagement"
+            ),
+            df[df.other_incentive == "TRUE"][["id", "created_at"]].assign(name="other"),
+        ]
+    ).rename(columns={"id": "challenge_id"})
+    incentives["name"] = pd.Categorical(
+        incentives["name"],
+        categories=["monetary", "publication", "speaking_engagement", "other"],
+    )
+    incentives = incentives.sort_values(["challenge_id", "name"])
+    incentives.index = np.arange(1, len(incentives) + 1)
+
+    # Challenge submission type(s)
+    sub_types = pd.concat(
+        [
+            df[df.file_submission == "TRUE"][["id", "created_at"]].assign(
+                name="prediction_file"
+            ),
+            df[df.container_submission == "TRUE"][["id", "created_at"]].assign(
+                name="container_image"
+            ),
+            df[df.notebook_submission == "TRUE"][["id", "created_at"]].assign(
+                name="notebook"
+            ),
+            df[df.mlcube_submission == "TRUE"][["id", "created_at"]].assign(
+                name="mlcube"
+            ),
+            df[df.other_submission == "TRUE"][["id", "created_at"]].assign(
+                name="other"
+            ),
+        ]
+    ).rename(columns={"id": "challenge_id"})
+    sub_types["name"] = pd.Categorical(
+        sub_types["name"],
+        categories=[
+            "prediction_file",
+            "container_image",
+            "notebook",
+            "mlcube",
+            "other",
+        ],
+    )
+    sub_types = sub_types.sort_values(["challenge_id", "name"])
+    sub_types.index = np.arange(1, len(sub_types) + 1)
+
+    return (
+        challenges.rename(columns={"platform": "platform_id"}).drop(
+            columns=["_platform"]
+        ),
+        incentives[["name", "challenge_id", "created_at"]],
+        sub_types[["name", "challenge_id", "created_at"]],
+    )
+
+
+def get_challenge_categories(
+    wks, sheet_name: str = "challenge_category"
```

Comment:
Does wks standard for "workspace"? I propose using the full name for the variable.

Reply:
`wks` was the variable name used in the `gspread` docs [[ref](https://docs.gspread.org/en/latest/index.html#quick-example)], so I just followed suit.  I believe it stands for spreadsheet or maybe worksheet?

Reply:
I'll add a type-hint for it to make it more clear.